nodefs/bridge.go

```diff
@@ -517,19 +525,120 @@ func (b *rawBridge) Fallocate(input *fuse.FallocateIn) (code fuse.Status) {
 }
 
 func (b *rawBridge) OpenDir(input *fuse.OpenIn, out *fuse.OpenOut) (status fuse.Status) {
-	return fuse.ENOSYS
+	n, _ := b.inode(input.NodeId, 0)
+	code := n.node.OpenDir(context.TODO())
+	if !code.Ok() {
+		return code
+	}
+	b.mu.Lock()
+	defer b.mu.Unlock()
+	out.Fh = b.registerFile(nil)
+	return fuse.OK
+}
+
+func (b *rawBridge) getStream(input *fuse.ReadIn, inode *Inode, f *fileEntry) fuse.Status {
+	if f.dirStream == nil || input.Offset == 0 {
+		if f.dirStream != nil {
+			f.dirStream.Close()
+		}
+		str, code := inode.node.ReadDir(context.TODO())
+		if !code.Ok() {
+			return code
+		}
+
+		f.hasOverflow = false
+		f.dirStream = str
```

Comment:
Does this need locking (or an atomic pointer swap?)

Reply:
I don't think so, because fuse/nodefs doesnt have locking there either.

Reply:
oops, there is locking, in fact.

But the race detector says nothing. hmm.